pkg/services/alerting/notifiers/telegram.go

```diff
@@ -107,17 +104,99 @@ func (this *TelegramNotifier) Notify(evalContext *alerting.EvalContext) error {
 		message = message + fmt.Sprintf("\n<i>Metrics:</i>%s", metrics)
 	}
 
-	bodyJSON.Set("text", message)
+	var body bytes.Buffer
+	w := multipart.NewWriter(&body)
+	fw, _ := w.CreateFormField("chat_id")
+	fw.Write([]byte(this.ChatID))
+
+	fw, _ = w.CreateFormField("parse_mode")
+	fw.Write([]byte("html"))
 
-	url := fmt.Sprintf(telegeramApiUrl, this.BotToken, "sendMessage")
-	body, _ := bodyJSON.MarshalJSON()
+	fw, _ = w.CreateFormField("text")
+	fw.Write([]byte(message))
 
+	w.Close()
+
+	this.log.Info("Sending telegram text notification", "chat_id", this.ChatID, "bot_token", this.BotToken)
+	url := fmt.Sprintf(telegramApiUrl, this.BotToken, "sendMessage")
 	cmd := &m.SendWebhookSync{
 		Url:        url,
-		Body:       string(body),
+		Body:       body.String(),
 		HttpMethod: "POST",
+		HttpHeader: map[string]string{
+			"Content-Type": w.FormDataContentType(),
+		},
 	}
 
+	return cmd
+}
+
+func (this *TelegramNotifier) buildImageMessage(evalContext *alerting.EvalContext) *m.SendWebhookSync {
```

Comment:
Too much code duplication in `buildImageMessage` and `buildTextMessage` 

Rather then having two functions doing almost the same thing. Have one function and add image fields if send image is set to true.

Reply:
The problem is that they are "almost the same thing", not "the same thing".

Differences:
* buildImageMessage needs to not embed HTML tags into the message (and it's called "caption" in the form data, not "message")
* one sends to sendPhoto, the other to sendMessage URI
* one embeds a file, the other does not

I think that merging them together will look subjectively uglier, but I'll do it as it's not a big deal, and you want that. I'll push an update this evening.